apps/base-docs/src/components/DocChat/docChat.ts

```diff
@@ -0,0 +1,257 @@
+      page_path: path,
+      action: ActionType.click,
+      component_type: ComponentType.button,
+    };
+
+    logEvent(type, updatedAttributes);
+  }
+};
+
+// Get Conversation ID
+export async function getConversationId(): Promise<number> {
+  let id: string = sessionStorage.getItem('BASE_AI_CONVERSATION_ID') ?? '';
+
+  if (!id) {
+    const response = await fetch('https://api.mendable.ai/v1/newConversation', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        api_key: '7f73a4b9-96b3-4689-8688-b120ea0ee916',
+      }),
+    });
+
+    const data: { conversation_id: string } = (await response.json()) as {
+      conversation_id: string;
+    };
+
+    if (data.conversation_id) {
+      id = data.conversation_id;
+      sessionStorage.setItem('BASE_AI_CONVERSATION_ID', id);
+
+      logGptEvent('gpt_conversation_created', {
+        conversation_id: parseInt(id),
+      });
+    }
+  }
+
+  return parseInt(id);
+}
+
+// Set and Get Session Storage Conversation
+export function setSessionConversation(conversation: ConversationMessage[]) {
+  const conversationString = JSON.stringify(conversation);
+  sessionStorage.setItem('BASE_AI_CONVERSATION', conversationString);
+}
+
+export function getSessionConversation(): ConversationMessage[] {
+  const conversationString: string = sessionStorage.getItem('BASE_AI_CONVERSATION') ?? '[]';
+
+  const conversation: ConversationMessage[] = JSON.parse(
+    conversationString,
+  ) as ConversationMessage[];
+
+  return conversation;
+}
+
+// POST Prompt and Stream Response
+export type ChatHistoryMessage = {
+  prompt: string;
+  response: string;
+};
+
+export type ConversationMessage = {
+  type: 'prompt' | 'response';
+  content: string;
+  sources?: string[];
+  helpful?: boolean | null;
+  messageId: number;
+};
+
+type ResponseSource = {
+  content: string;
+  data_id: string;
+  date_added: string;
+  id: number;
+  link: string;
+  manual_add: boolean;
+  relevance_score: number;
+  text: number;
+};
+
+export let controller: AbortController;
+
+export async function streamPromptResponse(
+  conversationId: number,
+  prompt: string,
+  setIsLoading: (isLoading: boolean) => void,
+  isGenerating: boolean,
+  setIsGenerating: (isGenerating: boolean) => void,
+  chatHistory: ChatHistoryMessage[],
+  setChatHistory: (chatHistory: (prevState: ChatHistoryMessage[]) => ChatHistoryMessage[]) => void,
+  setConversation: (
+    conversation: (prevState: ConversationMessage[]) => ConversationMessage[],
+  ) => void,
+) {
+  try {
+    let fullResponse = '';
+    let responseSources: ResponseSource[];
+    let messageId: number;
+
+    controller = new AbortController();
+
+    logGptEvent('gpt_prompt_submitted', {
+      conversation_id: conversationId,
+      prompt,
+    });
+
+    const url = 'https://api.mendable.ai/v1/mendableChat';
+
+    const data = {
+      api_key: '7f73a4b9-96b3-4689-8688-b120ea0ee916',
```

Comment:
Is this ok to have in here?

Reply:
Yep! This is a client-side API key that's protected by a domain whitelist configured in the Mendable dashboard. I'll be confirming the whitelist is set up properly before deploying this PR.